python/runNanoSeq.py

```diff
@@ -0,0 +1,938 @@
+
+#compute indels
+parser_indel = subparsers.add_parser('indel', help='indel caller')
+parser_indel.add_argument('-s','--sample', action='store', default='sample_1', help="sample name in output vcf (sample_1)")
+parser_indel.add_argument('--rb', type=int, action='store', default=2, help="minimum reads in a bundle. (2)")
+parser_indel.add_argument('--t3', type=int, action='store', default=135, help="excess bases above this value are trimmed from 3' (135)")
+parser_indel.add_argument('--t5', type=int, action='store', default=10, help="bases to trim from 5' reads (10)")
+parser_indel.add_argument('--mc', type=int, action='store', default=16, help="minimum bulk coverage (16)")
+
+#carry out gather operations and compute summaries
+parser_post = subparsers.add_parser('post', help='gather final files, compute summaries')
+
+args = parser.parse_args()
+
+#check job partition arguments
+if ( (args.index == None and args.max_index != None) or 
+     (args.index != None and args.max_index == None) ) :
+    raise ValueError( "Must specify index and max_index for array execution!")
+
+if ( args.index == None and args.max_index == None ) :
+  jobArray = False
+else :
+  jobArray = True
+
+if ( jobArray and args.threads > 1 ):
+  print("Warning: execution with a job array is single threaded so thread argument is ignored")
+
+#check the location of the tmpNanoSeq dir
+if not os.path.isdir(args.out) :
+  parser.error("Specified out directory %s is not accessible!"%args.out)
+
+try :
+  testfile = tempfile.TemporaryFile(dir = args.out)
+  testfile.close()
+except OSError :
+  sys.exit("\nCan't write to out directory %s\n"% args.out )
+
+#check required files
+if ( hasattr(args,'tumour') ) :
+  if not os.path.isfile(args.tumour) :
+    parser.error("BAM file %s was not found!" % args.tumour)
+  if not os.path.isfile(args.tumour + '.bai') :
+    parser.error("Index file %s was not found!" % (args.tumour + '.bai'))
+
+if ( hasattr(args,'normal') ) :
+  if not os.path.isfile(args.normal) :
+    parser.error("BAM file %s was not found!" % args.normal)
+  if not os.path.isfile(args.normal + '.bai') :
+    parser.error("Index file %s was not found!" % (args.normal + '.bai'))
+
+if ( hasattr(args,'ref') ) :
+  if not os.path.isfile(args.ref) :
+    parser.error("Reference file %s was not found!" % args.ref)
+  if not os.path.isfile(args.ref + '.fai') :
+    parser.error("Index file %s was not found!" % (args.ref + '.fai'))
+
+if ( hasattr(args,'snp') ) :
+  if not os.path.isfile(args.snp) :
+    parser.error("SNP file %s was not found!" % args.snp)
+  if not os.path.isfile(args.snp + '.tbi') :
+    parser.error("Index file %s was not found!" % (args.snp + '.tbi'))
+
+if ( hasattr(args,'mask') ) :
+  if not os.path.isfile(args.mask) :
+    parser.error("Mask file %s was not found!" % args.mask)
+  if not os.path.isfile(args.mask + '.tbi') :
+    parser.error("Index file %s was not found!" % (args.mask + '.tbi'))
```

Comment:
All of these can be compressed to a single function taking args:

* file
* idx_extension
* error prefix if really needed (BAM, SNP, Reference, Mask)

e.g.

```python3
def file_chk(fn, idx_ext, msg_prefix):
    if not os.path.isfile(fn) :
        parser.error(f"{msg_prefix} file {fn} was not found!")
    if not os.path.isfile(args.snp + '.tbi') :
        parser.error(f"{msg_prefix} index file {fn}{idx_ext} was not found!")

if ( hasattr(args,'tumour') ):
    file_chk(args.tumour, ".bai", "BAM")
if ( hasattr(args,'normal') ):
    file_chk(args.normal, ".bai", "BAM")
if ( hasattr(args,'ref') ):
    file_chk(args.ref, ".fai", "Reference")
if ( hasattr(args,'snp') ):
    file_chk(args.snp, ".tbi", "SNP")
if ( hasattr(args,'mask') ):
    file_chk(args.mask, ".tbi", "Mask")
```